deepmd/dpmodel/descriptor/base_descriptor.py

```diff
@@ -1,8 +1,78 @@
 # SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Callable,
+    Type,
+)
+
 import numpy as np
 
+from deepmd.utils.plugin import (
+    Plugin,
+)
+
 from .make_base_descriptor import (
     make_base_descriptor,
 )
 
-BaseDescriptor = make_base_descriptor(np.ndarray, "call")
+
+class BaseDescriptor(make_base_descriptor(np.ndarray, "call")):
```

Comment:
would it be better to move the plugin support to `make_base_descriptor`?

Reply:
I will try and see if there is anything wrong.